### PR TITLE
Add postprocessor for Rayleigh number calcs

### PR DESCRIPTION
**Pull request by GiudGiud**
Needed a new min/max postprocessor for functor material properties. 

It derives from Elementpostprocessor, but there could be an ElementFunctorPostprocessor intermediate class, same as there is for variables. What do you think?

### CODE REVIEW

**Comment by GiudGiud** [comment]
uh oh
https://www.google.com/search?q=thermal+diffusivity&oq=thermal+diffusi&aqs=chrome.0.0i433i512j69i57j0i512l8.4689j0j7&sourceid=chrome&ie=UTF-8

thermal diffusivity is not thermal conductivity :O

**Comment by GiudGiud** [comment]
Failure
    FAIL test:pressure_pulse.pressure_pulse_1d_adapt FAILED (CRASH)
is unrelated

**Comment by GiudGiud** [comment]
seems like a thread problem

**Comment by lindsayad** [comment]
That thread failure should have been solved by #20155

**Review by lindsayad — COMMENTED** [review]
This looks fine to me but I'm tired of resolving merge conflicts

**Comment by GiudGiud** [comment]
it can wait it's not needed right away

File: framework/src/postprocessors/ElementExtremeFunctorValue.C [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+template <bool is_ad>
+ElementExtremeFunctorValueTempl<is_ad>::ElementExtremeFunctorValueTempl(
+    const InputParameters & parameters)
+  : ElementPostprocessor(parameters),
+    _type((ExtremeType)(int)parameters.get<MooseEnum>("value_type")),
+    _functor(getFunctor<GenericReal<is_ad>>("functor")),
+    _proxy_functor(isParamValid("proxy_functor") ? getFunctor<GenericReal<is_ad>>("proxy_functor")
+                                                 : getFunctor<GenericReal<is_ad>>("functor"))
+{
```
Comment (GiudGiud) [inline framework/src/postprocessors/ElementExtremeFunctorValue.C:56]:
```suggestion
{
  if (isNodal())
    paramError("variable", "This AuxKernel only supports Elemental fields");
```

**Comment by GiudGiud** [comment]
@lindsayad this should be ready for review.
I ll merge my comment in at that time

File: framework/src/postprocessors/ElementExtremeFunctorValue.C [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  }
+}
+
+template <bool is_ad>
+void
+ElementExtremeFunctorValueTempl<is_ad>::computeValue()
+{
+  // Most element evaluations do not use skewness correction,
+  // but this could become a parameter in the future
+  Moose::ElemArg elem{_current_elem, false, false};
```
Comment (lindsayad) [inline framework/src/postprocessors/ElementExtremeFunctorValue.C:83]: Use `makeElemArg`?

File: modules/navier_stokes/include/postprocessors/RayleighNumber.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+{
+public:
+  static InputParameters validParams();
+
+  RayleighNumber(const InputParameters & parameters);
+
+protected:
+  virtual void initialize() override {}
+  virtual void execute() override {}
+  virtual Real getValue() override;
```
Comment (lindsayad) [inline modules/navier_stokes/include/postprocessors/RayleighNumber.h:28]: `override` always implies `virtual`, but this is fine

**Review by lindsayad — DISMISSED** [review]

**Comment by GiudGiud** [comment]
@lindsayad should be good now
This breaks VTB inputs so I ll have to go patch stuff.

File: framework/src/postprocessors/ElementExtremeFunctorValue.C [hunk 4]
```diff
@@ -80,7 +82,7 @@ ElementExtremeFunctorValueTempl<is_ad>::computeValue()
 {
   // Most element evaluations do not use skewness correction,
   // but this could become a parameter in the future
-  Moose::ElemArg elem{_current_elem, false, false};
+  Moose::ElemArg elem = makeElemArg(_current_elem);
```
Comment (GiudGiud) [inline framework/src/postprocessors/ElementExtremeFunctorValue.C:85]: could be a const i guess. next PR

Comment (lindsayad) [reply framework/src/postprocessors/ElementExtremeFunctorValue.C:85]: `const auto` indeed 😄

**Review by lindsayad — APPROVED** [review]

**Pull request merged**